cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionController.h

```diff
@@ -0,0 +1,311 @@
+   * converted as defined by the formatting hints given in option. This updates
+   * all textual representations owned by this controller.
+   * 
+   * The SpatialReference is taken from the GeoView.
+   * 
+   * If conversion fails this function is treated as a no-op.
+   * 
+   * \param point string representation of point. 
+   * \param option Option dictating hints on how to convert the string to a
+   *  point.
+   */
+  void setCurrentPoint(const QString& point, CoordinateConversionOption* option);
+
+  /*!
+   * \brief Set the current point to point. Point is a string which will be
+   * converted as defined by the formatting hints given in option. This updates
+   * all textual representations owned by this controller.
+   * 
+   * If conversion fails this function is treated as a no-op.
+   * 
+   * \param point string representation of point. 
+   * \param spatialReference An explicit SpatialReference for conversion.
+   * \param option Option dictating hints on how to convert the string to a
+   *  point.
+   */
+  void setCurrentPoint(const QString& point,
+                       const SpatialReference& spatialReference,
+                       CoordinateConversionOption* option);
+
+  /*!
+   * \brief Updates the GeoView's camera to point to the current point's
+   * location on the map.
+   * \sa zoomToDistance
+   */
+  void zoomToCurrentPoint();
+
+  /*!
+   * \brief Given option, generates a new result.
+   * 
+   * A new CoordianteConversionResult is added to the list-model returned by
+   * coordinateResults. This Result is tied to the conversion parameters as
+   * given by option.
+   * 
+   * \param option option to generate result for.
+   */
+  void addNewCoordinateResultForOption(CoordinateConversionOption* option);
+
+  /*!
+   * \brief Removes a given CoordinateConversionResult at index.
+   * 
+   * This function is for deleting results in the list-model returned by
+   * coordinateResults.
+   * 
+   * \param index index of a given CoordinateConversionResult in the list-model
+   * returned by coordinateResults.
+   */
+  void removeCoordinateResultAtIndex(int index);
+
+  /*!
+   * \internal
+   * \brief This is a helper function due to the inability to pass `Point` as
+   * a QML type. This forces this controller to emit the currentPointChanged
+   * signal again: which force all downstream consumers to update.
+   * 
+   * This is only an implementation detail so that the
+   * `CoordinateConversionOption` owned by the QML view can retrieve the
+   * current point when its format switches.
+   * 
+   * If Point were consumable by QML then we could just call the currentPoint
+   * getter instead.
+   */
+  void forceUpdateCoordinates();
+
+private:
+  Point m_currentPoint;
+  TaskWatcher m_screenToLocationProcess;
+  double m_zoomToDistance;
+  GenericListModel* m_coordinateFormats;
+  GenericListModel* m_conversionResults;
+  QObject* m_geoView;
+  bool m_inPickingMode;
```

Comment:
I would second all these recommendations to define these values in the header.

Reply:
2v1 here! I'll change.